core/src/main/java/cucumber/runtime/arquillian/client/CucumberArchiveProcessor.java

```diff
@@ -164,6 +169,20 @@ public void process(final Archive<?> applicationArchive, final TestClass testCla
         tryToAdd(libs, libraryContainer, "WEB-INF/lib/scala-library-", "cucumber.api.scala.ScalaDsl", "scala.App");
     }
 
+    private boolean archiveContains(Archive<?> archive, Class<?> clazz) {

```

Comment:
I tried it not to bound to a specific type of Archive, the WEB-INF is bound to `WebArchive` but the archive could also be a `JavaArchive` which will put its resources somewhere else.